### PR TITLE
Update raiden, fix typing problems

### LINKED ISSUE #744: Update services to latest raiden
Besides the usual updates this will require some typing changes, as web3 now exports its type definitions.

### PR DESCRIPTION
**Pull request by palango**
Fixes #744

### CODE REVIEW

File: src/raiden_libs/blockchain.py [hunk 1]
```diff
@@ -35,15 +36,31 @@
...
+    Currently works only with `HTTPProvider`. Needs to be adapted when new procviders
+    are added.
+    """
+    provider = web3.provider
+    if isinstance(provider, HTTPProvider):
+        endpoint = provider.endpoint_uri
+        if endpoint is not None:
+            return str(endpoint)
+
+    return ""
```
Comment (karlb) [inline src/raiden_libs/blockchain.py]: Should we raise an error here, instead?

File: tests/monitoring/monitoring_service/test_handlers.py [hunk 2]
```diff
@@ -169,7 +169,7 @@ def test_channel_opened_event_handler_adds_channel(context: Context):
 
 def test_channel_closed_event_handler_closes_existing_channel(context: Context):
     context = setup_state_with_open_channel(context)
-    context.web3.eth.blockNumber = BlockNumber(60)
+    context.web3.eth.blockNumber = BlockNumber(60)  # type: ignore
```
Comment (karlb) [inline tests/monitoring/monitoring_service/test_handlers.py]: What is the typing problem, here?

Comment (palango) [reply tests/monitoring/monitoring_service/test_handlers.py]: error: Property "blockNumber" defined in "Eth" is read-only

File: src/raiden_libs/blockchain.py (hunk 1 above)
Comment (palango) [reply src/raiden_libs/blockchain.py]: Good idea

**Pull request merged**